### ISSUE TITLE
ads version 0.12.3 not working with python 3.8

### ISSUE THREAD

**Issue by jvines** [andycasey/ads#126]
As the title says, I was trying out the example

```python
import ads.sandbox as ads
q = ads.SearchQuery(q='star')
for paper in q:
    print(paper.title, paper.citation_count)
```

## Expected Behavior
I should get a mock response from the ads

## Current Behavior
I get the following error:

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[15], line 1
----> 1 for paper in q:
      2     print(paper.title, paper.citation_count)

File ~/opt/anaconda3/lib/python3.8/site-packages/ads/search.py:501, in SearchQuery.__next__(self)
    498 # Allow immediate iteration without forcing a user to call .execute()
    499 # explicitly
    500 if self.response is None:
--> 501     self.execute()
    503 try:
    504     cur = self._articles[self.__iter_counter]

File ~/opt/anaconda3/lib/python3.8/site-packages/ads/sandbox.py:32, in SearchQuery.execute(self)
     30 def execute(self):
     31     with MockSolrResponse(SearchQuery.HTTP_ENDPOINT):
---> 32         super(SearchQuery, self).execute()

File ~/opt/anaconda3/lib/python3.8/site-packages/ads/search.py:533, in SearchQuery.execute(self)
    526 def execute(self):
    527     """
    528     Sends the http request implied by the self.query
    529     In addition, set up the request such that we can call next()
    530     to provide the next page of results
    531     """
    532     self.response = SolrResponse.load_http_response(
--> 533         self.session.get(self.HTTP_ENDPOINT, params=self.query)
    534     )
    536     # ADS will apply a ceiling to 'rows' and re-write the query
    537     # This code checks if that happened by comparing the reponse
    538     # "rows" with what we sent in our query
    539     # references https://github.com/andycasey/ads/issues/45
    540     recv_rows = int(self.response.responseHeader.get("params", {}).get("rows"))

File ~/opt/anaconda3/lib/python3.8/site-packages/requests/sessions.py:600, in Session.get(self, url, **kwargs)
    592 r"""Sends a GET request. Returns :class:`Response` object.
    593 
    594 :param url: URL for the new :class:`Request` object.
    595 :param \*\*kwargs: Optional arguments that ``request`` takes.
    596 :rtype: requests.Response
    597 """
    599 kwargs.setdefault("allow_redirects", True)
--> 600 return self.request("GET", url, **kwargs)

File ~/opt/anaconda3/lib/python3.8/site-packages/requests/sessions.py:587, in Session.request(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)
    582 send_kwargs = {
    583     "timeout": timeout,
    584     "allow_redirects": allow_redirects,
    585 }
    586 send_kwargs.update(settings)
--> 587 resp = self.send(prep, **send_kwargs)
    589 return resp

File ~/opt/anaconda3/lib/python3.8/site-packages/requests/sessions.py:701, in Session.send(self, request, **kwargs)
    698 start = preferred_clock()
    700 # Send the request
--> 701 r = adapter.send(request, **kwargs)
    703 # Total elapsed time of the request (approximately)
    704 elapsed = preferred_clock() - start

File ~/opt/anaconda3/lib/python3.8/site-packages/requests/adapters.py:489, in HTTPAdapter.send(self, request, stream, timeout, verify, cert, proxies)
    487 try:
    488     if not chunked:
--> 489         resp = conn.urlopen(
    490             method=request.method,
    491             url=url,
    492             body=request.body,
    493             headers=request.headers,
    494             redirect=False,
    495             assert_same_host=False,
    496             preload_content=False,
    497             decode_content=False,
    498             retries=self.max_retries,
    499             timeout=timeout,
    500         )
    502     # Send the request.
    503     else:
    504         if hasattr(conn, "proxy_pool"):

File ~/opt/anaconda3/lib/python3.8/site-packages/urllib3/connectionpool.py:703, in HTTPConnectionPool.urlopen(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)
    700     self._prepare_proxy(conn)
    702 # Make the request on the httplib connection object.
--> 703 httplib_response = self._make_request(
    704     conn,
    705     method,
    706     url,
    707     timeout=timeout_obj,
    708     body=body,
    709     headers=headers,
    710     chunked=chunked,
    711 )
    713 # If we're going to release the connection in ``finally:``, then
    714 # the response doesn't need to know about the connection. Otherwise
    715 # it will also try to release it and we'll have a double-release
    716 # mess.
    717 response_conn = conn if not release_conn else None

File ~/opt/anaconda3/lib/python3.8/site-packages/urllib3/connectionpool.py:386, in HTTPConnectionPool._make_request(self, conn, method, url, timeout, chunked, **httplib_request_kw)
    384 # Trigger any extra validation we need to do.
    385 try:
--> 386     self._validate_conn(conn)
    387 except (SocketTimeout, BaseSSLError) as e:
    388     # Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.
    389     self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)

File ~/opt/anaconda3/lib/python3.8/site-packages/urllib3/connectionpool.py:1042, in HTTPSConnectionPool._validate_conn(self, conn)
   1040 # Force connect early to allow us to validate the connection.
   1041 if not getattr(conn, "sock", None):  # AppEngine might not have  `.sock`
-> 1042     conn.connect()
   1044 if not conn.is_verified:
   1045     warnings.warn(
   1046         (
   1047             "Unverified HTTPS request is being made to host '%s'. "
   (...)
   1052         InsecureRequestWarning,
   1053     )

File ~/opt/anaconda3/lib/python3.8/site-packages/urllib3/connection.py:419, in HTTPSConnection.connect(self)
    410 if (
    411     not self.ca_certs
    412     and not self.ca_cert_dir
   (...)
    415     and hasattr(context, "load_default_certs")
    416 ):
    417     context.load_default_certs()
--> 419 self.sock = ssl_wrap_socket(
    420     sock=conn,
    421     keyfile=self.key_file,
    422     certfile=self.cert_file,
    423     key_password=self.key_password,
    424     ca_certs=self.ca_certs,
    425     ca_cert_dir=self.ca_cert_dir,
    426     ca_cert_data=self.ca_cert_data,
    427     server_hostname=server_hostname,
    428     ssl_context=context,
    429     tls_in_tls=tls_in_tls,
    430 )
    432 # If we're using all defaults and the connection
    433 # is TLSv1 or TLSv1.1 we throw a DeprecationWarning
    434 # for the host.
    435 if (
    436     default_ssl_context
    437     and self.ssl_version is None
    438     and hasattr(self.sock, "version")
    439     and self.sock.version() in {"TLSv1", "TLSv1.1"}
    440 ):

File ~/opt/anaconda3/lib/python3.8/site-packages/urllib3/util/ssl_.py:449, in ssl_wrap_socket(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)
    437     warnings.warn(
    438         "An HTTPS request has been made, but the SNI (Server Name "
    439         "Indication) extension to TLS is not available on this platform. "
   (...)
    445         SNIMissingWarning,
    446     )
    448 if send_sni:
--> 449     ssl_sock = _ssl_wrap_socket_impl(
    450         sock, context, tls_in_tls, server_hostname=server_hostname
    451     )
    452 else:
    453     ssl_sock = _ssl_wrap_socket_impl(sock, context, tls_in_tls)

File ~/opt/anaconda3/lib/python3.8/site-packages/urllib3/util/ssl_.py:493, in _ssl_wrap_socket_impl(sock, ssl_context, tls_in_tls, server_hostname)
    490     return SSLTransport(sock, ssl_context, server_hostname)
    492 if server_hostname:
--> 493     return ssl_context.wrap_socket(sock, server_hostname=server_hostname)
    494 else:
    495     return ssl_context.wrap_socket(sock)

File ~/opt/anaconda3/lib/python3.8/ssl.py:500, in SSLContext.wrap_socket(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)
    494 def wrap_socket(self, sock, server_side=False,
    495                 do_handshake_on_connect=True,
    496                 suppress_ragged_eofs=True,
    497                 server_hostname=None, session=None):
    498     # SSLSocket class handles server_hostname encoding before it calls
    499     # ctx._wrap_socket()
--> 500     return self.sslsocket_class._create(
    501         sock=sock,
    502         server_side=server_side,
    503         do_handshake_on_connect=do_handshake_on_connect,
    504         suppress_ragged_eofs=suppress_ragged_eofs,
    505         server_hostname=server_hostname,
    506         context=self,
    507         session=session
    508     )

File ~/opt/anaconda3/lib/python3.8/ssl.py:1004, in SSLSocket._create(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)
    999 kwargs = dict(
   1000     family=sock.family, type=sock.type, proto=sock.proto,
   1001     fileno=sock.fileno()
   1002 )
   1003 self = cls.__new__(cls, **kwargs)
-> 1004 super(SSLSocket, self).__init__(**kwargs)
   1005 self.settimeout(sock.gettimeout())
   1006 sock.detach()

TypeError: super(type, obj): obj must be an instance or subtype of type
```

And if I use ads instead of the sandbox I get this error:

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[19], line 2
      1 for paper in q:
----> 2     print(paper.title, paper.citation_count)

File ~/opt/anaconda3/lib/python3.8/site-packages/ads/utils.py:30, in cached_property.__get__(self, obj, type)
     23 value = obj.__dict__.get(self.__name__, _missing)
     24 if value is _missing:
     25     # One time warning that the user is using lazy loading
     26     warnings.warn(
     27         "You are lazy loading attributes via '{}', and so are "
     28         "making multiple calls to the API. This will impact your overall "
     29         "rate limits."
---> 30         .format(self.func.__name__),
     31         UserWarning,
     32     )
     33     value = self.func(obj)
     34     obj.__dict__[self.__name__] = value

AttributeError: 'cached_property' object has no attribute 'func'
```

## Your Environment
I'm using Python 3.8 and ads version 0.12.3

**Comment by keflavich** [comment]
This isn't a python 3.8 issue, but a werkzeug issue.  It's solved in https://github.com/andycasey/ads/pull/117

**Comment by andycasey** [comment]
#117 is merged and a new version (0.12.4) is tagged, released, and uploaded to PyPI. Can you please try that version? If you're still having the same issue then please re-open this issue.
